### DISCUSSION TITLE
Cannot restore backup

### DISCUSSION THREAD

**Discussion by Ariveraq** [jgyates/genmon#1087] [Questions] [answered]
### Steps to Reproduce

When I initially installed the application in my PI 3B I first tried using terminal but was unable to make it work.  For some reason when I connected to the PI via terminal and was asked for my password and entered the password I got access denied.  I decided to use Putty, using the same credentials I tried on terminal I was able to access the PI.  I installed the application and everything was working fine.  I changed the SD card for a A10 card (had it installed on an old card I had) and reinstalled the application on the new card, before I did the new install I made a backup.  I am trying to restore the backup using the procedure on the appendix. 
     
https://github.com/jgyates/genmon/wiki/Appendix-M-Tips-and-Tricks-useful-for-Genmon

To transfer the file from my PC I tried Filezilla, I am able to connect to the PI but when I do the transfer I get access denied.  Because  terminal does not work I am unable to execute the "scp genmon_backup.tar.gz pi@192.168.1.10:./" command.  

Why do I get access denied with terminal and not from Putty?
What is preventing Filezilla to access the SD card?

Thanks in advance!

### Expected Result

Perform a backup restoration. 

### Actual Result

Unable.

### Version Info

Version V1.19.02

**Comment by jgyates** [comment]
I am converting this issue to a discussion question as this is related to pi setup and not a bug or feature request.

**Comment by jgyates** [comment]
you mentioned that you were "using terminal". Can you give more details on what this is exactly. There are a few things named terminal (a MacOS terminal session, a direct connected serial console terminal, a pi with keyboard and monitor attached). Provide the exact steps you are taking to connect "using terminal" a this is not descriptive enough to allow me to provide any feedback.

filezilla is a FTP client, you must have a FTP server enabled on the pi for this to work.

Also, for scp or ssh to work you must have the SSH server running on the pi. 

PuTTY can use several protocols to acces the pi (SCP, SSH, Telnet, rlogin, and raw socket connection. It can also connect to a serial port). I don't know which one you are using.

If you can log into the pi you can enable the ssh server by typing:

     sudo raspi-config

Then select Interfacing Options then select SSH, then enable. You may need to restart your pi once you enable the SSH server.

How did you set your password for the pi? Did you set it with the raspberry pi imager program?

Also, this thread may be of some use: https://github.com/jgyates/genmon/issues/1015

**Comment by Ariveraq** [comment]
For terminal I use "Windows PowerShell".  When I tried using ssh genmon.local it did not work.  It only works with IP.
The SSH server is enabled on the PI, I am able to connect with Filezilla to the PI.  For some reason when I try the transfer I get access denied.
With Putty I am using SSH, I am able to log in with genmon.local or using the IP.
The password was set with the PI imager.

![Filezilla](https://github.com/jgyates/genmon/assets/163960230/1317f62d-1420-4acf-b341-2ef75a8b1565)
![Putty](https://github.com/jgyates/genmon/assets/163960230/9ff130d1-cc50-453f-b541-84e1f4229bea)
![Terminal](https://github.com/jgyates/genmon/assets/163960230/4e3a3c9b-b4e5-4fa3-9daa-3c14c51e104d)

**Reply by jgyates** [reply, marked as answer]
Your problem is that you are using different user names with ssh in a power shell vs putty. The user name you use must match the user name on the pi and the password for that user on the pi. With the ssh command if you do not specify the user name on the command like it will use the user name of the current logged in user on the windows system. Notice when you are running ssh in power shell, you type

      ssh 192.168.68.65

You did not include a user name so the ssh program tried to log you into the raspberry pi with the username alexi, which I assume is the user name on the windows system.

With Putty, there is a default user name under Connection->Data. Use that user name with ssh and scp

     ssh username@192.168.68.65

or

      scp genmon_backup.tar.gz username@192.168.1.10:./ 

The command: 
    
     scp genmon_backup.tar.gz pi@192.168.1.10:./

assumes a username of 'pi' on the raspberry pi. If you setup a different username than pi, you must use that username.

Let me know if that resolves you issue.,

**Reply by Ariveraq** [reply]
Thanks for the clarification,  I am new to this.  On the procedure to restore the backup, you first transfer the file from the PC to the PI then you login into the PI and unpack the backup, correct?  How would this command scp genmon_backup.tar.gz username@192.168.1.10:./  knows where the backup is in the pc?

**Reply by jgyates** [reply]
yes, the order you listed is correct.

The scp command is run from your PC. MacOS has scp built in, but if you installed the ssh client on windows then it likely comes with scp as well. You run the scp command from the same folder that the file genmon_backup.tar.gz is located. For example to change folders from your home folder to the Downloads folder, in the power shell type

     cd Downloads

If the file genmon_backup.tar.gz is in this folder then you can type:

     scp genmon_backup.tar.gz username@192.168.1.10:./ 

This will transfer the genmon_backup.tar.gz file located in the current directory to the pi in the home folder (the './" at the end is the home folder) of the user specified by username.

**Comment by markman1000** [comment]
I have a question about using genmon on a Briggs and Stratton generator.  From what I have read it looks like a can purchase a RS 485 to wifi adaptor and connect it to the Briggs controller.  The terminal strip is easy to get to.  Then I seems like I then use the wife to get it into the Raspberry Pi computer.  Then do I use the same code as for the Generic generator.  I really appreciate what you have done in this space.  Briggs and Stratton sucks in providing monitering.

Mark
________________________________

**Comment by jgyates** [comment]
the code is the same, however the settings are different. The serial interface can still use the onboard raspberry pi port howerver you  will need to use a RS-485 adapter instead of a RS-232 adapter.

This page has info on setup and equipment:

https://github.com/jgyates/genmon/wiki/Appendix-P-Briggs-and-Stratton-Controller-Information

The link also has links to videos related to the Briggs and Stratton setup.

**Comment by Ariveraq** [comment]
Mr. Yates, happy to report that I was able to restore my backup.  Thank you very much for your support and dedication to this project.

**Comment by jgyates** [comment]
I am glad it is working for you.

**Comment by markman1000** [comment]
I have a question and I don’t really understand how to reach you without just replying to one of your emails.

 

I just bought a cabin that has a Briggs and Stratton backup generator and has a 1032 controller.  Since we only live here part time I would like to be able to monitor it.  My other house has a Kolher generator and I was able to buy a package from Kolher, install it and start monitoring the generator.  It was straight forward and took less than an hour.

 

I have talked to Briggs and Stratton and they are a piece of work.  They will not talk to me and refer me to one of their dealers.  I have made repeated requests, and they won’t return my call.

 

That is when I ran across Genmon and it looks exactly like what I am looking for.

 

I am a retired mechanical engineer, and I was aware of raspberry pi.  I have been following your Github site.  I am not sure I have the brain cells left to learn how to use and program Raspberry pi.  I have opened the generator and I see the connections so that part is easy.

 

Do you or anyone you know do this work on a contract basis.  I would be happy to supply the equipment and pay someone to program it and get it ready to install.

 

Sorry about the long email and thanks for your time.

 

Mark

**Comment by jgyates** [comment]
@markman1000 

The best way to ask questions is to create a new discussion thread here: https://github.com/jgyates/genmon/discussions

Regarding getting a pi setup, the easiest route would be to purchase a pre-configured pi from pintsize.me. The purchase page is here: https://pintsize.me/store/ols/products/generator-hats-and-phat

You want a pre-configured HAT with the pi 3. This will work with the Briggs and Stratton controller, however you only need to make a cable connection and setup the controller. The Briggs instructions are located here: https://github.com/jgyates/genmon/wiki/Appendix-P-Briggs-and-Stratton-Controller-Information

Regarding a contractor, you probably want someone local who can make the need cable and and enclosure for the pi. The above pre-configured pi will have all the software loaded you need. I am not currently doing any contract programming.

There are sites like https://www.fiverr.com/ that you can locate people for hire who may be able to help in your area.

**Reply by skipfire** [reply]
@jgyates the HAT itself would not be needed correct?

Could a RS485 to USB (something like https://www.amazon.com/dp/B076WVFXN8/ without the breakout part) work for the cable instead of assembling one?  With the other recent change to the install process allowing a user to select USB that may be a cleaner & easier hardware setup.

**Comment by jgyates** [comment]
@skipfire 

You are correct, the HAT is not needed. Thanks for catching that. I forgot the Briggs and Stratton uses RS485 instead of RS232

@markman1000, you can use the link above that @skipfire references or one of the solutions referenced in the wiki page: https://github.com/jgyates/genmon/wiki/Appendix-P-Briggs-and-Stratton-Controller-Information#rs-485-information
